### DISCUSSION TITLE
Changing File Path When Creating a File in a Virtual Drive

### DISCUSSION THREAD

**Discussion by chienbg2000** [dokan-dev/dokan-dotnet#350] [Q&A] [answered]
Hello, I am creating a virtual drive from the directory structure in SQLite, storing the addresses of files in a folder with their name being their ID in SQLite. How can I customize the location to save a file with a custom name and path when a user pastes a file into my virtual drive, please?

**Comment by Liryna** [comment]
Hi @chienbg2000 , this is error prone but when the item is created, you can lie that it is successfully created and after a moment of inactivity, make it disappear and reappear at location you want.

**Reply by chienbg2000** [reply]
Thank you very much. I have a question, in the sample about mirror, which part of the code specifies where the file will be saved? I only see codes retrieving files from mirrorPath.

**Reply by Liryna** [reply]
Oh you previously meant where the virtual content is actually saved. [GetFilePath](https://github.com/dokan-dev/dokany/blob/master/samples/dokan_mirror/mirror.c#L86-L90) is used to map the virtual path to the local path

**Reply by chienbg2000** [reply]
in this code at https://github.com/dokan-dev/dokan-dotnet/blob/master/sample/DokanNetMirror/Mirror.cs, I can't find the part of the code that specifies where the file is saved

**Reply by Liryna** [reply]
GetPath is the function for the C# sample. The path is then used by the standard C# api for create / write etc

**Reply by chienbg2000** [reply]
you mean ? 
![Screenshot 2024-03-14 090127](https://github.com/dokan-dev/dokan-dotnet/assets/57701993/37a63b64-4c27-4e15-960c-ada35f1c0a13)
but why is there a file in filePath before the file is created

**Reply by Liryna** [reply, marked as answer]
Yes and the file might actually be created by this function depending on the parameters

**Reply by chienbg2000** [reply]
Thank you very much

**Reply by chienbg2000** [reply]
I am using Dokan with SQLite to create a virtual folder. When my SQLite data changes, what do I need to do to refresh the folder in Dokan?

**Reply by Liryna** [reply]
You can use [DokanNet.Dokan.Notify](https://dokan-dev.github.io/dokan-dotnet-doc/html/class_dokan_net_1_1_dokan_1_1_notify.html) to replay the changes in the virtual drive which will make explorer refresh the view.
